### PR TITLE
Remove setup-node cache

### PR DESCRIPTION
**Pull request by McGiverGim**
As stated in the setup-node action documentation, the cache is used to store **global** node_modules dependencies.

As far as I know, we are not using those dependencies. We use only local node_modules dependencies, that we are caching manually in the CI.

This PR removes the cache of the setup-node to speed up a little the process.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
